server/src/main/java/org/candlepin/model/PoolFilterBuilder.java

```diff
@@ -218,56 +227,105 @@ private DetachedCriteria createPoolAttributeCriteria(String attribute, List<Stri
+        String originalPoolAlias = this.alias.isEmpty() ? "this." : this.alias;
+        subquery.add(Property.forName(originalPoolAlias + "id").eqProperty("PoolAttrI.pool_id"));
+        // subquery.setProjection(Projections.property("attr.id"));
+        subquery.setProjection(Projections.id());
 
         return subquery;
     }
 
+    @SuppressWarnings("checkstyle:indentation")
     private DetachedCriteria createProductAttributeCriteria(String attribute, List<String> values) {
         DetachedCriteria subquery = DetachedCriteria.forClass(Pool.class, "PoolI")
             .createAlias("PoolI.product", "ProdI")
             .createAlias("ProdI.attributes", "ProdAttrI");
 
-        subquery.add(new FilterLikeExpression("ProdAttrI.name", attribute, false));
+        subquery.add(new FilterLikeExpression("ProdAttrI.indices", attribute, false))
+            .add(Restrictions.sqlRestriction(
+                "NOT EXISTS (SELECT poolattr.pool_id FROM cp_pool_attribute poolattr " +
+                "WHERE poolattr.pool_id = this_.id AND LOWER(poolattr.name) LIKE LOWER(?) ESCAPE '!')",
+                this.sanitizeMatchesFilter(attribute), StringType.INSTANCE
+            ));
 
         Disjunction disjunction = Restrictions.disjunction();
         for (String value : values) {
             if (value == null || value.isEmpty()) {
-                disjunction.add(Restrictions.isNull("ProdAttrI.value"));
-                disjunction.add(Restrictions.eq("ProdAttrI.value", ""));
+                disjunction.add(Restrictions.isNull("ProdAttrI.elements"));
+                disjunction.add(Restrictions.eq("ProdAttrI.elements", ""));
             }
             else {
-                disjunction.add(new FilterLikeExpression("ProdAttrI.value", value, true));
+                disjunction.add(new FilterLikeExpression("ProdAttrI.elements", value, true));
             }
         }
 
         subquery.add(disjunction);
 
         String originalPoolAlias = this.alias.isEmpty() ? "this." : alias;
         subquery.add(Property.forName(originalPoolAlias + "id").eqProperty("PoolI.id"));
-        subquery.setProjection(Projections.property("PoolI.id"));
+        // subquery.setProjection(Projections.property("PoolI.id"));
+        subquery.setProjection(Projections.id());
 
         // We don't want to match Product Attributes that have been overridden
-        DetachedCriteria overridden = DetachedCriteria.forClass(PoolAttribute.class, "PoolAttrI")
-            // If we're using wildcards in the name, we should block exact matches
-            .add(Restrictions.eqProperty("PoolAttrI.name", "ProdAttrI.name"))
-            .add(Restrictions.eqProperty("PoolI.id", "PoolAttrI.pool.id"))
-            .setProjection(Projections.property("PoolAttrI.pool.id"));
-        subquery.add(Subqueries.notExists(overridden));
+
+        // TODO: FIXME: This is broken, the entire filter builder needs to be rewritten to properly
```

Comment:
Technically no, it's no longer used. But it should be fixed/removed as part of the overarching cleaning that needs to happen in PoolCurator and EntitlementCurator.